src/circuit/gadget/poseidon.rs

```diff
@@ -149,13 +151,13 @@ impl<
     pub fn absorb(
         &mut self,
         mut layouter: impl Layouter<F>,
-        value: Word<F, PoseidonChip, S, T, RATE>,
+        value: CellValue<F>,
```

Comment:
For symmetry, we should squeeze out `CellValue`s as well, keeping `Word`s for state cells, and the only guarantee we provide for the squeezed cells would be that they are in equality-constrained columns.

Non-blocking, as this is just as easy for me to do when addressing #222.